### PR TITLE
IP Logic Remaining Calories

### PR DESCRIPTION
**Pull request by skorpiom**
Hey,

It's logic for remaining calories component. 
Please review.

### CODE REVIEW

File: Test/SetRemianingCalories.test.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        duration: 60,
+        calories: 686
+      }]
+  } ];
+
+const userAge = getAge(testUser.dateOfBirth);
+const goalCalories = Math.floor(calculateCalories(testUser.gender, testUser.weights[0].weight, testUser.height, userAge, testUser.activityLevel));
+
+localStorage[KEY_LOGGED_USER] = '';
+localStorage[loggedInUser] = '';
```
Comment (robert-czwartosz) [inline Test/SetRemianingCalories.test.ts:60]: Why loggedInUser is cleared from local storage?
I think that `localStorage[KEY_LOGGED_USER] = '';` is sufficient.

File: Test/SetRemianingCalories.test.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+describe('set remaining calories component test', () => {
+
+  test('no logged user in local storage',()=>{
+    expect(SetRemainingCalories()).toBe(undefined);
+  });
+  test('logged user - no user object in ls',()=>{
+    localStorage[KEY_LOGGED_USER] = loggedInUser;
```
Comment (robert-czwartosz) [inline Test/SetRemianingCalories.test.ts:68]: I suggest to use functions you implemented (setLoggedInUser, getLoggedInUser and clearLoggedInUser).

File: Test/SetRemianingCalories.test.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    expect(SetRemainingCalories()).toBe(undefined);
+  });
+  test('logged user and set correct user object in ls',()=>{
+    localStorage[KEY_LOGGED_USER] = loggedInUser;
+    localStorage[loggedInUser] = JSON.stringify(testUser);
```
Comment (robert-czwartosz) [inline Test/SetRemianingCalories.test.ts:73]: I think
`setLoggedInUser(loggedInUser);`
`saveToLocalStorage(loggedInUser, testUser);`
looks more naturally.

File: Test/SetRemianingCalories.test.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    expect(SetRemainingCalories()).toBeInstanceOf(HTMLElement);
+  });
+  test('diaryExercises correct date and provided kcal',()=>{
+    const component = SetRemainingCalories();
+    const nodeListRows = component.querySelectorAll('.remaining-calories-tile__section-details--table tr td');
+    expect(nodeListRows[2].textContent).toEqual(`${testUser.diaryExercises[0].totalCalories}`);
+  });
+  test('no diaryExercises field (diaryFood works the same)',()=>{
+    delete testUser.diaryExercises;
+    localStorage[loggedInUser] = JSON.stringify(testUser);
```
Comment (robert-czwartosz) [inline Test/SetRemianingCalories.test.ts:83]: `saveToLocalStorage(loggedInUser, testUser);`

File: Test/SetRemianingCalories.test.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    const component = SetRemainingCalories();
+    const nodeListRows = component.querySelectorAll('.remaining-calories-tile__section-details--table tr td');
+    expect(nodeListRows[2].textContent).toEqual("0");
+  });
+  test('no diaryExercises.date field (diaryFood works the same)',()=>{
+    testUser.diaryExercises = [];
+    localStorage[loggedInUser] = JSON.stringify(testUser);
```
Comment (robert-czwartosz) [inline Test/SetRemianingCalories.test.ts:90]: saveToLocalStorage...

**Review by robert-czwartosz — COMMENTED** [review]
Looks great!
Only there are small issues with code.

**Review by fszwedo — APPROVED** [review]
nothing from me to be changed :)

File: Test/SetRemianingCalories.test.ts (hunk 1 above)
Comment (skorpiom) [reply Test/SetRemianingCalories.test.ts:60]: Actually both are redundant. Local storage at the beginning is empty (during testing).

I cleared `localStorage[loggedInUser] = '';` bcs it's an object with users data and logic also has specified behavior in that situation.

File: Test/SetRemianingCalories.test.ts (hunk 2 above)
Comment (skorpiom) [reply Test/SetRemianingCalories.test.ts]: You're right, it's easier to read it later

File: Test/SetRemianingCalories.test.ts (hunk 3 above)
Comment (skorpiom) [reply Test/SetRemianingCalories.test.ts]: done

File: Test/SetRemianingCalories.test.ts (hunk 4 above)
Comment (skorpiom) [reply Test/SetRemianingCalories.test.ts]: done

File: Test/SetRemianingCalories.test.ts (hunk 5 above)
Comment (skorpiom) [reply Test/SetRemianingCalories.test.ts]: done

**Review by robert-czwartosz — APPROVED** [review]

**Comment by robert-czwartosz** [comment]
After resolving conflict in package.json, some test doesn't pass.

**Comment by robert-czwartosz** [comment]
Bug is in test file.
I'll fix it and merge.

**Review by robert-czwartosz — APPROVED** [review]

**Pull request merged**